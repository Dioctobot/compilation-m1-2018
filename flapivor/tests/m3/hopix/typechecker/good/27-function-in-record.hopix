type F<a, b> = { apply : a -> b }

def f : Int -> Int (x) = x + 1

val boxed_f : F<Int, Int> = { apply = f }<Int, Int>

def apply_boxed_f : forall a, b. F<a, b> * a -> b (boxed_f, a) = boxed_f.apply (a)

val main : Int = apply_boxed_f<Int, Int>(boxed_f, 42)