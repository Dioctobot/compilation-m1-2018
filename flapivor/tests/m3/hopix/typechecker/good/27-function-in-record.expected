f : Int -> Int
boxed_f : F<Int, Int>
apply_boxed_f : forall a, b. F<a, b> * a -> b
main : Int
